File: leetcode/maximum-depth-of-n-ary-tree/myHan.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    private int walk(Node cur, int step) {
        if(cur == null) return step;
        int ret = step + 1;
        for(int i = 0; i < cur.children.size(); i++) {
            ret = Math.max(ret, walk(cur.children.get(i), step+1));
        }
        return ret;
    }
    public int maxDepth(Node root) {
        return walk(root, 0);
    }
}

File: leetcode/4-hamming-distance/myHan.cc
class Solution {
public:
    int hammingDistance(int x, int y) {
        int z = x ^ y;
        int ret = 0;
        for(int i = 0; z >> i; i++) {
            ret += (z >> i) & 1;
        }
        return ret;
    }
};